staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -98,6 +98,46 @@ func ValidateUpdateCustomResourceDefinitionStatus(obj, oldObj *apiextensions.Cus
 	return allErrs
 }
 
+// ValidateCustomResourceDefinitionVersion statically validates
+func ValidateCustomResourceDefinitionVersion(version *apiextensions.CustomResourceDefinitionVersion, fldPath *field.Path, index int) field.ErrorList {
+	allErrs := field.ErrorList{}
+	subresourcesInvalid := false
+	if version.Subresources != nil {
+		if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceWebhookConversion) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("subresources"), "disabled by feature-gate CustomResourceWebhookConversion"))
+			subresourcesInvalid = true
+		}
+		if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceSubresources) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("subresources"), "disabled by feature-gate CustomResourceSubresources"))
```

Comment:
this means that a CRD created with an enabled gate (and defaulting) will become un-updatable after the gate is disabled. This is another hint that the conditional defaulting is wrong.

Reply:
Note: that CRD did not make use of conversion webhooks or of subresources per version. So it should keep working even after disabling the gate.

Reply:
It's updatable if a user PUTs a valid object with nil per-version fields, because during update we validate the new CRD spec and the immutable fields, but not the entire old CRD spec. 

For a patch request that updates the top level fields, the per-version fields read from storage will fail the validation. The defaulter should prune the per-version fields if the gate is disabled. This is same as rolling-back to 1.11 which doesn't understand the per-version fields.